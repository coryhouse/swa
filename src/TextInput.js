import React from "react";
import PropTypes from "prop-types";
import createReactClass from "create-react-class";

const TextInput = createReactClass({
  render: function() {
    const { label, id, onChange, value, name, showLabel } = this.props;
    return (
      <div>
        {showLabel && <label htmlFor={id}>{label}</label>}
        <br />
        <input
          id={id}
          type="text"
          name={name}
          onChange={onChange}
          value={value}
        />
      </div>
    );
  }
});

TextInput.propTypes = {
  showLabel: PropTypes.bool,
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

TextInput.defaultProps = {
  showLabel: true
};

export default TextInput;
