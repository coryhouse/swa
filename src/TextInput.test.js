import React from "react";
import TextInput from "./TextInput";
import renderer from "react-test-renderer";

it("renders label by default", () => {
  const tree = renderer
    .create(
      <TextInput
        id="firstName"
        label="First Name"
        name="firstName"
        value="example"
        onChange={jest.fn()}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("hides label when showLabel is false", () => {
  const tree = renderer
    .create(
      <TextInput
        showLabel={false}
        id="firstName"
        label="First Name"
        name="firstName"
        value="example"
        onChange={jest.fn()}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
