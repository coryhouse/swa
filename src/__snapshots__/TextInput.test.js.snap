// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hides label when showLabel is false 1`] = `
<div>
  <br />
  <input
    id="firstName"
    name="firstName"
    onChange={[MockFunction]}
    type="text"
    value="example"
  />
</div>
`;

exports[`renders label by default 1`] = `
<div>
  <label
    htmlFor="firstName"
  >
    First Name
  </label>
  <br />
  <input
    id="firstName"
    name="firstName"
    onChange={[MockFunction]}
    type="text"
    value="example"
  />
</div>
`;
